pkg/pillar/hypervisor/kubevirt.go

```diff
@@ -0,0 +1,1421 @@
+	// FixedResources.Memory is in Kbytes
+	memoryLimit := strconv.Itoa(config.Memory * 1000)
+	memoryRequest := strconv.Itoa(config.Memory * 1000)
+
+	pod := &k8sv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        kubeName,
+			Namespace:   kubeapi.EVEKubeNameSpace,
+			Annotations: annotations,
+		},
+		Spec: k8sv1.PodSpec{
+			Containers: []k8sv1.Container{
+				{
+					Name:            kubeName,
+					Image:           ociName,
+					ImagePullPolicy: k8sv1.PullNever,
+					SecurityContext: &k8sv1.SecurityContext{
+						Privileged: &[]bool{true}[0],
+					},
+					Resources: k8sv1.ResourceRequirements{
+						Limits: k8sv1.ResourceList{
+							k8sv1.ResourceCPU:    resource.MustParse(vcpus),
+							k8sv1.ResourceMemory: resource.MustParse(memoryLimit),
+						},
+						Requests: k8sv1.ResourceList{
+							k8sv1.ResourceCPU:    resource.MustParse(vcpus),
+							k8sv1.ResourceMemory: resource.MustParse(memoryRequest),
+						},
+					},
+				},
+			},
+			DNSConfig: &k8sv1.PodDNSConfig{
+				Nameservers: []string{"8.8.8.8", "1.1.1.1"}, // XXX, temp, Add your desired DNS servers here
+			},
+		},
+	}
+	pod.Labels = make(map[string]string)
+	pod.Labels[eveLabelKey] = domainName
+	logrus.Infof("CreatePodConfig: pod setup %+v", pod)
+
+	// Now we have VirtualMachine Instance object, save it to config file for debug purposes
+	// and save it in context which will be used to start VM in Start() call
+	meta := vmiMetaData{
+		pod:      pod,
+		isPod:    true,
+		name:     kubeName,
+		domainId: int(rand.Uint32()),
+	}
+	ctx.vmiList[domainName] = &meta
+
+	podStr := fmt.Sprintf("%+v", pod)
+
+	// write to config file
+	file.WriteString(podStr)
+
+	return nil
+}
+
+func encodeSelections(selections []netattdefv1.NetworkSelectionElement) string {
+	bytes, err := json.Marshal(selections)
+	if err != nil {
+		logrus.Errorf("encodeSelections %v", err)
+		return ""
+	}
+	return string(bytes)
+}
+
+func StartPodContiner(kubeconfig *rest.Config, pod *k8sv1.Pod) error {
+
+	clientset, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		logrus.Errorf("StartPodContiner: can't get clientset %v", err)
+		return err
+	}
+
+	opStr := "created"
+	_, err = clientset.CoreV1().Pods(kubeapi.EVEKubeNameSpace).Create(context.TODO(), pod, metav1.CreateOptions{})
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			// TODO: update
+			logrus.Errorf("StartPodContiner: pod create filed: %v", err)
+			return err
+		} else {
+			opStr = "already exists"
+		}
+	}
+
+	logrus.Infof("StartPodContiner: Pod %s %s with nad %+v", pod.ObjectMeta.Name, opStr, pod.Annotations)
+
+	err = checkForPod(kubeconfig, pod.ObjectMeta.Name)
+	if err != nil {
+		logrus.Errorf("StartPodContiner: check for pod status error %v", err)
+		return err
+	}
+	logrus.Infof("StartPodContiner: Pod %s running", pod.ObjectMeta.Name)
+	return nil
+}
+
+func checkForPod(kubeconfig *rest.Config, podName string) error {
+	var i int
+	var status string
+	var err error
+	for {
```

Comment:
What is the reason to check 6 times? It sounds arbitrary.

Moreover why is an endless loop needed?

Reply:
yes, we should put some comments around, and maybe defined a const for the total wait time on this.

Reply:
Thx. That sounds good.

Reply:
@naiming-zededa your response needed here.

Reply:
ack. will be in next commit

Reply:
Note that we shouldn't have to worry about watchdogs here. See other comments.